### PR TITLE
Update go version to 1.20

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Update go version to match k/k 1.27: https://github.com/kubernetes/kubernetes/blob/master/.go-version

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Update go version to 1.20.2 for k/k 1.27
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 ARG BUILDPLATFORM
 
-FROM --platform=$BUILDPLATFORM golang:1.19.6 as builder
+FROM --platform=$BUILDPLATFORM golang:1.20.2 as builder
```
Comment (msau42) [inline Dockerfile:17]: It looks like 1.20.3 is being released today with a number of CVE fixes:
https://groups.google.com/g/golang-announce/c/71Wg3N0IZk0/m/xchF2bEZAgAJ

Comment (sunnylovestiramisu) [reply Dockerfile]: Updated.

**Pull request merged**